sdks/java/io/elasticsearch/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTest.java

```diff
@@ -0,0 +1,303 @@
+    node = NodeBuilder.nodeBuilder().settings(settingsBuilder).build();
+    LOGGER.info("Elasticsearch node created");
+    node.start();
+  }
+
+  @Before
+  public void before() throws Exception {
+    IndicesAdminClient indices = node.client().admin().indices();
+    IndicesExistsResponse indicesExistsResponse =
+        indices.exists(new IndicesExistsRequest(ES_INDEX)).get();
+    if (indicesExistsResponse.isExists()) {
+      indices.prepareDelete(ES_INDEX).get();
+    }
+  }
+
+  private List<String> createDocuments(long nbDocs, boolean insertErrors){
+    String[] scientists =
+        { "Einstein", "Darwin", "Copernicus", "Pasteur", "Curie", "Faraday", "Newton", "Bohr",
+            "Galilei", "Maxwell" };
+    ArrayList<String> data = new ArrayList<>();
+    for (int i = 0; i < nbDocs; i++) {
+      int index = i % scientists.length;
+      //insert 2 malformed documents is asked to do so
+      if (insertErrors && (i == 6 || i == 7)){
+        data.add(String.format("{\"scientist\";\"%s\", \"id\":%d}", scientists[index], i));
+      } else {
+        data.add(String.format("{\"scientist\":\"%s\", \"id\":%d}", scientists[index], i));
+      }
+    }
+    return data;
+  }
+  private void insertTestDocuments(long nbDocs) throws Exception {
+    Client client = node.client();
+    final BulkRequestBuilder bulkRequestBuilder = client.prepareBulk().setRefresh(true);
+    List<String> data = createDocuments(nbDocs, false);
+    for (String document:data){
+      bulkRequestBuilder.add(client.prepareIndex(ES_INDEX, ES_TYPE, null).setSource(document));
+    }
+    final BulkResponse bulkResponse = bulkRequestBuilder.execute().actionGet();
+    if (bulkResponse.hasFailures()) {
+      throw new IOException(String.format("Cannot insert test documents in index %s : %s",
+                                          ES_INDEX, bulkResponse.buildFailureMessage()));
+    }
+  }
+
+  @Test
+  public void testSizes() throws Exception {
+    insertTestDocuments(NB_DOCS);
+    PipelineOptions options = PipelineOptionsFactory.create();
+    ElasticsearchIO.Read read =
+        ElasticsearchIO.read().withConnectionConfiguration(connectionConfiguration);
+    BoundedElasticsearchSource initialSource =
+        new BoundedElasticsearchSource(read, null);
+    // can't use equal assert as Elasticsearch indexes never have same size
+    // (due to internal Elasticsearch implementation)
+    long estimatedSize = initialSource.getEstimatedSizeBytes(options);
+    LOGGER.info("Estimated size: {}", estimatedSize);
+    assertTrue("Wrong estimated size", estimatedSize > 40000);
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testRead() throws Exception {
+    insertTestDocuments(NB_DOCS);
+
+    TestPipeline pipeline = TestPipeline.create();
+
+    PCollection<String> output = pipeline.apply(
+        ElasticsearchIO.read().withConnectionConfiguration(connectionConfiguration)
+                              .withScrollKeepalive("5m"));
```

Comment:
No, the test would not fail because default 1m is enough to process 1 document. So you're right withScrollKeepalive("5m") is not critical for this test, I'll remove it if it is misleading about the test's intention.